queries/shellscript.scm

```diff
@@ -267,82 +267,45 @@
 ;; Names, values, and types
 ;;
 
-;;!! foo="bar"
-;;!  ^^^
-;;!  xxxx
-;;!  ---------
+;;!! for ((i = 1; i <= 5; i++)); do
+;;!        ^
+;;!        xxxx
+;;!        -----
 (
   (variable_assignment
-    name: (_) @name @_.trailing.start.startOf
-    .
-    "=" @_.trailing.end.endOf
+    name: (_) @name @value.leading.endOf
+    value: (_) @value @name.trailing.startOf
   ) @dummy @_.domain
   (#not-parent-type? @dummy declaration_command)
 )
 
-;;!! for ((i = 1; i <= 5; i++)); do
-;;!        ^
-;;!        xxxx
-;;!        -----
-(variable_assignment
-  !name
-  (variable_name) @name @_.trailing.start.startOf
-  "="
-  (_) @_.trailing.end.startOf
-) @_.domain
-
 ;;!! local foo="bar"
 ;;!        ^^^
 ;;!  xxxxxxxxxx
 ;;!  ---------------
 (declaration_command
-  "local" @_.domain.start.startOf @_.trailing.start.startOf
   (variable_assignment
     name: (_) @name
-    .
-    "=" @_.trailing.end.endOf
-  ) @_.domain.end.endOf
-)
-
-;;!! foo="bar"
-;;!      ^^^^^
-;;!     xxxxxx
-;;!  ---------
-(
-  (variable_assignment
-    "=" @value.leading.start.startOf
-    .
-    value: (_) @value @value.leading.end.endOf
-  ) @dummy @_.domain
-  (#not-parent-type? @dummy declaration_command)
-)
+    value: (_) @_.removal.end.startOf
+  )
+) @_.domain @_.removal.start.startOf
```

Comment:
I don't think I'd ever seen/noticed `.removal`. Much nicer :D

Reply:
Yeah we generally prefer leading / trailing if it makes sense, but note that the behaviour of leading / trailing may not be as you expect. During removal, we only remove one of them. The canonical example is the comma in a comma-separated list: you don't want to remove both commas

If you need to remove something from both sides, removal is the way to go